xbmc/application/ApplicationPowerHandling.h

```diff
@@ -0,0 +1,87 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "addons/Addon.h"
+#ifdef TARGET_WINDOWS
+#include "powermanagement/WinIdleTimer.h"
+#endif
+#include "utils/Stopwatch.h"
+#include "windowing/OSScreenSaver.h"
+
+#include <string>
+
+class CApplicationPlayer;
+
+class CApplicationPowerHandling
+{
+public:
+  explicit CApplicationPowerHandling(CApplicationPlayer& appPlayer);
+
+  bool IsScreenSaverInhibited() const;
+  void ResetScreenSaver();
+  void SetScreenSaverLockFailed() { m_iScreenSaveLock = -1; }
+  void SetScreenSaverUnlocked() { m_iScreenSaveLock = 1; }
+  void StopScreenSaverTimer();
+
+  int GlobalIdleTime();
+  void ResetSystemIdleTimer();
+
+  void ResetShutdownTimers();
+  void StopShutdownTimer();
+
+  bool ToggleDPMS(bool manual);
+
+  // Wakes up from the screensaver and / or DPMS. Returns true if woken up.
+  bool WakeUpScreenSaverAndDPMS(bool bPowerOffKeyPressed = false);
+
+protected:
```

Comment:
I assume the goal here is just to remove stuff from CApplication as an interim solution right?
Most of the stuff here likely belongs to GUI (or at least I see them as owned somehow by CGUIComponent)

Reply:
some of it is somewhat interim yes, a lot of stuff that sits in CApplication can probably be moved to saner places. but before any such reorganization can be done we need an overview of dependencies within the huge capp class. this is imo a way to do that in a step-wise fashion, and in itself having value as things get compartmentalized (even though it's all pulled back into the monolith through inheritance). here you can now clearly see that the only thing that ties the powerhandling to the rest of the CApplication class is the appPlayer member which is passed by ref in the constructor.

Reply:
Yeap that's what I thought :+1: 
Thanks much for going through the pain of cleaning this gigantic piece of crap

Reply:
Indeed. The first step is to identify dependencies by seeing how far we can separate the code.